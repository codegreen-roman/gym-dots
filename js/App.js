import React from 'react'

const now = new Date()

// This is a stateless functional component
export const App = () => (
    <section className='app-container'>
        <div>Current time is:</div>
        <div>{now.toLocaleTimeString()}</div>
    </section>
)
