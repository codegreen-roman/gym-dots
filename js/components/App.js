import React from 'react'
import { Header } from './header/Header'
import { Footer } from './footer/Footer'
import { Route } from 'react-router-dom'
import { Activity } from './Activity'
import { Status } from './Status'
import { Exercises } from './Exercises'
import moment from 'moment'

export const App = () => {

    const dateStr = moment().format('dddd, MMM Do')
    const headerTitle = 'Start Workout'

    return (
        <section className='app-container'>
            <Header dateStr={dateStr} subTitle={headerTitle} />
            <Route exact path='/' component={() => <Activity />} />
            <Route path='/status' component={() => <Status />} />
            <Route path='/list' component={() => <Exercises />} />
            <Footer />
        </section>
    )
}
