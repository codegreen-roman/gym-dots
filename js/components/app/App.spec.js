import React from 'react'
import configureMockStore from 'redux-mock-store'
import { shallow } from 'enzyme'

import { App } from './App'

const setup = () => {
    const store = configureMockStore()()
    const wrapper = shallow(<App store={store} />)

    return {
        store,
        wrapper
    }
}

describe('App', () => {
    test('renders without crashing', () => {
        const { wrapper } = setup()
        expect(wrapper).toMatchSnapshot()
    })
})
