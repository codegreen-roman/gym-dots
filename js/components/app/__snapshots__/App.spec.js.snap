// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<section
  className="app-container"
>
  <Switch>
    <Route
      exact={true}
      path="/welcome"
    >
      <Connect(_Timer)>
        <div>
          Child 1
        </div>
        <div>
          Child 2
        </div>
        <div>
          Child 3
        </div>
        <div>
          Child 4
        </div>
      </Connect(_Timer)>
    </Route>
    <Route
      component={[Function]}
      path="/"
    />
    <Route
      component={[Function]}
    />
  </Switch>
</section>
`;
